### ISSUE TITLE
Cairo requires libXt development files

### ISSUE THREAD

**Issue by glin** [rstudio/r-system-requirements#31]
The Cairo R package has an additional dependency on libXt development headers (`libxt-dev` on Ubuntu or `libXt-devel` on CentOS), which isn't declared in the SystemRequirements: https://cran.r-project.org/web/packages/Cairo/index.html

Without libXt, Cairo installations will fail from missing an `X11/Intrinsic.h` file:
```
xlib-backend.c:34:74: fatal error: X11/Intrinsic.h: No such file or directory
 #include <X11/Intrinsic.h>      /*->    Xlib.h  Xutil.h Xresource.h .. */
                                                                          ^
compilation terminated.
make: *** [xlib-backend.o] Error 1
ERROR: compilation failed for package 'Cairo'
```
https://github.com/cran/Cairo/blob/34248669a392e7d33a5ac23f3784c6eb57f4dcac/src/xlib-backend.c#L34

There's been some discussion on removing X Intrinsics as a dependency for Cairo? but I don't think it's happened yet: https://lists.cairographics.org/archives/cairo/2014-August/025521.html

Since Cairo is a fairly common dependency for R packages, we should address this as soon as we add support for CRAN package overrides. Or consider adding libXt dev to the cairo rule as a temporary stopgap.

**Comment by zhijunyuu** [comment]
I have exactly the same issue as you, could you share how did you solve the problem ?

**Comment by pitkant** [comment]
As this issue is still open and I had exactly the same problem as of 2022, I would like to point out that tips from the following SO question helped me install Cairo (on Ubuntu 20.4.5 LTS, rocker/r-ver:4.2.1): https://stackoverflow.com/questions/23642353/error-message-installing-cairo-package-in-r

Basically adding `libcairo2-dev libxt-dev xvfb xauth xfonts-base libgtk2.0-dev` to `RUN apt-get update && apt-get install`. Note that not all of those might be necessary to successfully installing Cairo but I included them all to be on the safe side.

Also seems to affect RHEL #104

**Comment by glin** [comment]
Yup, looks like the Cairo R package still requires libXt even though it's not declared. At this point, we might as well add libXt to the Cairo rule to work around this just because Cairo is so heavily used and depended on. There are a few other Cairo-using packages like httpgd/gdtools that don't need libXt, but getting Cairo working probably make it a worthy tradeoff.

**Comment by glin** [comment]
I asked about this in the Cairo issue tracker: https://github.com/s-u/Cairo/issues/39. libXt was apparently unneeded and removed from Cairo!

So we'll just wait for the next release of Cairo on CRAN and then close this issue. If the CRAN release takes too long, we could also still add libXt as a temporary stopgap.
